workflows/pipe-common/shell/install_nfs_client

```diff
@@ -23,31 +23,86 @@ IS_RPM_BASED=$?
+    read -r -d '' INSTALL_LUSTRE_CMD <<- EOM
+      apt-get install wget -y -qq  &&
+      wget -q $LUSTRE_CLIENT_URL -O lustre-client.tar.gz &&
+      mkdir lustre-client-install/ &&
+      tar -C lustre-client-install/ -zxvf lustre-client.tar.gz &&
+      mkdir -p /lib/modules/$(uname -r) &&
+      dpkg -i lustre-client-install/* &&
+      rm -rf lustre-client-install/ lustre-client.tar.gz
+EOM
 fi
 
-eval "$CHECK_CMD"
+eval "$CHECK_LUSTRE_CMD"
+IS_LUSTRE_INSTALLED=$?
+
+eval "$CHECK_NFS_SMB_CMD"
+IS_NFS_SMB_INSTALLED=$?
 
 ######################################################
 # If NFS client is already installed - skip installation, otherwise install
 ######################################################
-if [ $? -ne 0 ]
+if [ $IS_NFS_SMB_INSTALLED -eq 0 ] && [ $IS_LUSTRE_INSTALLED -eq 0 ]
 then
+    pipe_log_info "--> NFS client is already installed" "$NFS_INSTALL_TASK"
+else
     pipe_log_info "--> NFS client not found, proceeding with installation" "$NFS_INSTALL_TASK"
-    pipe_log_info "--> Installing NFS client" "$NFS_INSTALL_TASK"
-    eval "$INSTALL_CMD"
-    if [ $? -ne 0 ]
+    pipe_log_info "--> Installing NFS clients" "$NFS_INSTALL_TASK"
+    if [ $IS_NFS_SMB_INSTALLED -ne 0 ]
     then
-        pipe_log_fail "Failed to install NFS client, process will not continue with shared FS initialization" "$NFS_INSTALL_TASK"
-        exit 1
+        pipe_log_info "--> Installing [nfs, smb] client" "$NFS_INSTALL_TASK"
+        eval "$INSTALL_NFS_SMB_CMD"
+        if [ $? -ne 0 ]
+        then
+            pipe_log_fail "Failed to install NFS client [nfs, smb], process will not continue with shared FS initialization" "$NFS_INSTALL_TASK"
+            exit 1
+        else
+            pipe_log_info "--> NFS client [nfs, smb] installed successfully" "$NFS_INSTALL_TASK"
+        fi
+    else
+        pipe_log_info "--> NFS client [nfs, smb] installed already" "$NFS_INSTALL_TASK"
     fi
-    pipe_log_info "--> NFS client installed" "$NFS_INSTALL_TASK"
-else
-    pipe_log_info "--> NFS client is already installed" "$NFS_INSTALL_TASK"
+    if [ $IS_LUSTRE_INSTALLED -ne 0 ]
+    then
+        pipe_log_info "--> Installing [lustre] client" "$NFS_INSTALL_TASK"
+        LUSTRE_INSTALLATION_LOG="/run/lustre_client_installation.log"
+        eval "$INSTALL_LUSTRE_CMD" > $LUSTRE_INSTALLATION_LOG 2>&1
+        if [ $? -ne 0 ]
+        then
+            pipe_log_fail "Errors occured during NFS client [lustre] installation, check $LUSTRE_INSTALLATION_LOG for more details" "$NFS_INSTALL_TASK"
+        else
+            pipe_log_info "--> NFS client packages [lustre] installed successfully" "$NFS_INSTALL_TASK"
+        fi
+    else
+        pipe_log_info "--> NFS client packages [lustre] installed already" "$NFS_INSTALL_TASK"
+    fi
+    eval "df -t lustre" > /dev/null 2>&1
+    if [[ -z $(cat /proc/filesystems | grep lustre) ]]; then
+        pipe_log_warn "--> [lustre] filesystem is not available. " "$NFS_INSTALL_TASK"
+    fi
```

Comment:
I suppose there is no need to explicitly check for lustre availability. If lustre doesn't work in container then the corresponding shares mounting will just fail anyway with more precise logs.

Reply:
Done